adapters/python/sources/message_dump_player.py

```diff
@@ -0,0 +1,162 @@
+import signal
+import time
+from typing import Optional, Tuple
+
+import msgpack
+from savant_rs.utils.serialization import (
+    Message,
+    load_message_from_bytes,
+)
+from savant_rs.zmq import BlockingWriter, WriterConfigBuilder
+
+from adapters.shared.thread import BaseThreadWorker
+from savant.utils.config import opt_config, strtobool
+from savant.utils.config import req_config
+from savant.utils.logging import get_logger, init_logging
+from savant.utils.welcome import get_starting_message
+
+LOGGER_NAME = 'adapters.message_dump_player'
+
+logger = get_logger(LOGGER_NAME)
+
+
+class Config:
+    """Configuration for the adapter."""
+
+    def __init__(self):
+        self.playlist_path = req_config('PLAYLIST_PATH')
+        self.zmq_endpoint = req_config('ZMQ_ENDPOINT')
+        self.sync_output = opt_config('SYNC_OUTPUT', False, strtobool)
+
+
+class MessageDumpReader:
+    """Reads messages from the message dump file."""
+
+    def __init__(self, file_path: str):
+        self._list_file = open(file_path, 'rb')
+        self._message_dump_file = None
+        self._unpacker = None
+        self._next_message_dump_file()
+
+    def read(self) -> Optional[Tuple[int, str, Message, bytes]]:
+        """Unpack the next message from the file."""
+
+        if not self._unpacker:
+            return None
+        message = None
+        while message is None:
+            try:
+                message = self._unpacker.unpack()
+            except msgpack.OutOfData:
+                if not self._next_message_dump_file():
+                    return None
+        ts, topic, meta, content = message
+
+        return ts, bytes(topic).decode(), load_message_from_bytes(meta), content
+
+    def _next_message_dump_file(self):
+        """Open the next message dump file from the list."""
+
+        if self._message_dump_file:
+            self._message_dump_file.close()
+
+        file_path = self._list_file.readline()
+        if not file_path:
+            return False
+
+        self._message_dump_file = open(file_path.strip(), 'rb')
+        self._unpacker = msgpack.Unpacker(self._message_dump_file)
+
+        return True
+
+    def __del__(self):
+        if self._message_dump_file:
+            self._message_dump_file.close()
+        if self._list_file:
+            self._list_file.close()
+
+
+class Egress(BaseThreadWorker):
```

Comment:
If it is from another adapter, try reusing it if it is possible. Not sure how it differs. Maybe it is not straightforward, then ignore this.

Reply:
They have the writer in common, and it's possible to refactor the buffer to update the workload method and use it here with a different logic for receiving messages, but it seemed to me not enough to generalize, so I could rename it to Player to avoid ambiguity

Reply:
ok